cmake/Autoconfig.cmake

```diff
@@ -26,6 +26,19 @@ else()
   check_support(HAS_STACKWALK has_stackwalk.cpp "" "dbghelp" "")
 endif()
 
+if(CMAKE_CXX_STANDARD GREATER_EQUAL 20)
+  # check_cxx_source_compiles doesn't support modules (yet?) so we need to drop
+  # to a raw try_compile.
+  try_compile(
+    HAS_CXX20_MODULES
```

Comment:
It looks like this test will pass for much older versions of clang than I was expecting, all the way back to clang 10 while I'm under the impression the module implementation only got more feature-complete and stable recently https://gcc.godbolt.org/z/4r73zGxd6. On the other hand, it won't work for any version of gcc because of `-fmodules` (which might be ok, not sure how stable gcc is yet but I'm under the impression they're at least close to feature-complete / stability). Thoughts on how best to approach this? I'm not sure what the best default behavior is.

Reply:
I'm thinking compiler version might be more robust yeah (but more of a hassle to check)